### PR TITLE
Remove `TmpDaemon`, use `Daemon` instead

### PR DESCRIPTION
**Pull request by jcredding**
This removes the temporary placeholder `TmpDaemon` that was used
in place of `Daemon` and switches to using `Daemon` directly. This
was needed for the `ConfigFile` and `Process` related parts of
Qs. There were implemented using the `TmpDaemon` but now that the
`Daemon` has been implemented it can be used instead. This will
allow qs to actually run a daemon and process jobs.

@kellyredding - Ready for review.

### CODE REVIEW

**Comment by kellyredding** [comment]
@jcredding did you mean "_They_ were implemented using" in the commit msg?

File: test/unit/daemon_tests.rb [hunk 1]
```diff
@@ -160,6 +160,12 @@ class InitTests < InitSetupTests
       assert_instance_of configuration.logger.class, data.logger
     end
 
+    should "know its name and pid file" do
+      data = subject.daemon_data
+      assert_equal data.name,     subject.name
+      assert_equal data.pid_file, subject.pid_file
+    end
+
```
Comment (kellyredding) [inline test/unit/daemon_tests.rb]: @jcredding are you missing "should have" regression method tests for these?  And are you missing them for all the imeths?  Not sure but I can't find them.

**Comment by kellyredding** [comment]
@jcredding a comment about some possibly missing tests.  Otherwise :-1: .

File: test/unit/daemon_tests.rb (hunk 1 above)
Comment (jcredding) [inline test/unit/daemon_tests.rb]: @kellyredding - Yep they were missing and I added them. Nice catch.

**Pull request merged**